waltz-client/src/main/java/com/wepay/waltz/client/internal/InternalStreamClient.java

```diff
@@ -90,7 +90,7 @@ public TransactionBuilderImpl getTransactionBuilder(TransactionContext context)
         Partition partition = getPartition(context.partitionId(numPartitions));
         partition.ensureMounted();
 
-        return new TransactionBuilderImpl(partition.nextReqId(), callbacks.getClientHighWaterMark(partition.partitionId));
+        return new TransactionBuilderImpl(partition.nextReqId(), partition.clientHighWaterMark());
```

Comment:
`partition.clientHighWaterMark()` reads the current client high-water mark from `AtomicLong` in the partition object. It is incremented immediately after `applyTransaction()`.